File: python/raindrops/raindrops.py
"""The raindrops challenge on the exercism website."""

sounds = {
    3: "Pling",
    5: "Plang",
    7: "Plong",
}


def convert(number: int) -> str:
    """Produce raindrop sounds based on factors.

    :param number: int - the input number.
    :return: str - added sounds or the number itself."""
    result = ""
    for factor, sound in sounds.items():
        if number % factor == 0:
            result += sound
    if result:
        return result
    else:
        return str(number)

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

var soundMap = map[int]string{
	3: "Pling",
	5: "Plang",
	7: "Plong",
}

func Convert(number int) string {
	var sound string = ""

	// DO NOT DO THIS!
	// Iteration order over a map is not guaranteed to be deterministic!
	// https://go.dev/blog/maps#iteration-order
	for k, v := range soundMap {
		if number%k == 0 {
			sound += v
		}
	}
	if sound == "" {
		sound = fmt.Sprintf("%d", number)
	}
	return sound
}
